clippy_lints/src/default_numeric_fallback.rs

```diff
@@ -80,14 +81,23 @@ impl<'a, 'tcx> NumericFallbackVisitor<'a, 'tcx> {
                             LitKind::Int(_, LitIntType::Unsuffixed) | LitKind::Float(_, LitFloatType::Unsuffixed));
                 if !ty_bound.is_numeric();
                 then {
-                    let suffix = match lit_ty.kind() {
-                        ty::Int(IntTy::I32) => "i32",
-                        ty::Float(FloatTy::F64) => "f64",
+                    let (suffix, is_float) = match lit_ty.kind() {
+                        ty::Int(IntTy::I32) => ("i32", false),
+                        ty::Float(FloatTy::F64) => ("f64", true),
                         // Default numeric fallback never results in other types.
                         _ => return,
                     };
 
-                    let sugg = format!("{}_{}", snippet(self.cx, lit.span, ""), suffix);
+                    let src = if let Some(src) = snippet_opt(self.cx, lit.span){
+                        src
+                    } else {
+                        match lit.node  {
+                            LitKind::Int(src, _) =>format!("{}", src),
+                            LitKind::Float(src, _) => format!("{}", src),
+                            _ => { return; }
+                        }
```

Comment:
Formattings:

```suggestion
                    let src = if let Some(src) = snippet_opt(self.cx, lit.span) {
                        src
                    } else {
                        match lit.node {
                            LitKind::Int(src, _) => format!("{}", src),
                            LitKind::Float(src, _) => format!("{}", src),
                            _ => return,
                        }
```

Reply:
Just a question:
Could you tell me how do you find formatting problems inside `if_chain`? Is there a good way to find them inside macros?

Reply:
I don't know of anything that would help here. Maybe implementing formatting rules in `rustfmt` for the if_chain macro. Not sure if that would be accepted there though, since this isn't a std macro.

Reply:
I understand. Thanks for the answer!